docs/src/datamanagment/metadata/metadata-search.md

```diff
@@ -0,0 +1,324 @@
+| user_metadata      | no      | Map<string, string> | User-defined metadata (e.g., annotations, tags). If none exists, an empty map is stored. |
+| committer          | yes     | string              | The user who committed the object’s latest change.                                                                                                     |
+| content_type       | no      | string              | The MIME type of the object (e.g., `application/json`, `image/png`).                                                                                                                                                     |
+| creation_date      | yes     | Timestamp           | The original creation timestamp of the object in the repository.                                                                                                                             |
+
+!!! info
+    lakeFS object metadata tables are eventually consistent, which means it may take up to a few minutes for newly committed 
+    objects to become searchable. Metadata becomes searchable **atomically** — either all object metadata from the commit 
+    is available, or none of it is. 
+
+!!! tip 
+    To check whether the most recent state of your branch is available for metadata search queries, check if the following 
+    query returns results:
+    ```sql
+    USE "<repo>-metadata.<branch>.system";
+    SELECT * FROM object_metadata
+    WHERE commit_id = <head_commit> -- Replace with the head commit ID of the branch you are looking at 
+    LIMIT 1;
+    ```
+
+## Getting Started
+
+This section assumes that you are already using lakeFS [object metadata](../../understand/glossary.md#object-metadata). 
+
+## Configuring Metadata Search
+
+TODO
+* metadata server configurations
+  * lakeFS server
+  * Searchable repos and branches
+* Iceberg catalog configurations?
+
+## How to Search by Metadata
+
+To search by object metadata in lakeFS, you query the Iceberg object metadata tables lakeFS creates and manages.
+
+Object metadata tables are standard Iceberg tables, meaning, you can query it using any Iceberg-compatible engine, including
+DuckDB, Trino, Spark, PyIceberg, or else.
+
+!!! tip
+    Query performance depends heavily on the engine you use. For example, Trino typically delivers better performance than PyIceberg.
+
+If you're using DuckDB, note the Iceberg REST Catalog [guide](../../integrations/iceberg.md#relative-namespace-support) for 
+details on how to reference `object_metadata` tables.
+
+#### Search Steps
+
+To run a metadata search query:
+1. [Initialize](../../integrations/iceberg.md#catalog-initialization-example-using-pyiceberg) lakeFS Iceberg catalog, and authenticate.
+2. Load the object metadata table that represents the reference you would like to query.
+3. Use SQL to search by system or user-defined metadata.
+
+Here’s an example using PyIceberg and DuckDB: 
```

Comment:
note that the code doesn't import duckdb, I assume it requires the duckdb package to be installed for this integration. consider note about the packages to install in order for the code to run or create one example with python code that uses pyiceberg and one that uses duckdb directly.